
# 7.1 异常概述与异常体系结构

![title](https://raw.githubusercontent.com/XJZ-0707/imge/master/gitnote/2019/09/24/%E5%BC%82%E5%B8%B8%E5%8E%9F%E5%9B%A0-1569293452289.jpg)

![title](https://raw.githubusercontent.com/XJZ-0707/imge/master/gitnote/2019/09/24/%E5%BC%82%E5%B8%B8-1569293526714.jpg)
![title](https://raw.githubusercontent.com/XJZ-0707/imge/master/gitnote/2019/09/24/%E5%BC%82%E5%B8%B8%E4%BD%93%E7%B3%BB-1569294477301.jpg)

![title](https://raw.githubusercontent.com/XJZ-0707/imge/master/gitnote/2019/09/24/%E5%BC%82%E5%B8%B8%E4%BD%93%E7%B3%BB2-1569294564438.jpg)
## Java虚拟机无法解决的严重问题
```java 
package com.day16.java;
/*
 * Error:
 * Java虚拟机无法解决的严重问题。如：JVM系统内部错误、资源耗尽等严重情况。比如：StackOverflowError和OOM。
 * 
 * 一般不编写针对性的代码进行处理。
 * 
 * 
 */
public class ErrorTest {
	public static void main(String[] args) {
		//1.栈溢出：java.lang.StackOverflowError
//		main(args);
		//2.堆溢出：java.lang.OutOfMemoryError 
		Integer[] arr = new Integer[1024*1024*1024];
	}
}

```


# 7.2 常见异常


# 7.3 异常处理机制一：try-catch-finally


# 7.4 异常处理机制二：throws


# 7.5 手动抛出异常：throw


# 7.6 用户自定义异常类
